x-pack/plugins/ml/public/file_datavisualizer/components/edit_flyout/overrides_validation.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+const INDETERMINATE_FIELD_PLACEHOLDER = '?';
+const FRACTIONAL_SECOND_SEPARATORS = ':.,';
+
+const VALID_LETTER_GROUPS = {
+  'yyyy': true,
+  'yy': true,
+  'M': true,
+  'MM': true,
+  'MMM': true,
+  'MMMM': true,
+  'd': true,
+  'dd': true,
+  'EEE': true,
+  'EEEE': true,
+  'H': true,
+  'HH': true,
+  'h': true,
+  'mm': true,
+  'ss': true,
+  'a': true,
+  'XX': true,
+  'XXX': true,
+  'zzz': true,
+};
+// TODO: ensure this is the best way to do this check
+function isLetter(str) {
+  return str.length === 1 && str.match(/[a-z]/i);
+}
+
+export function isTimestampFormatValid(timestampFormat) {
+  const result = { isValid: true, errorMessage: null };
+
+  if (timestampFormat.indexOf(INDETERMINATE_FIELD_PLACEHOLDER) >= 0) {
+    result.isValid = false;
+    result.errorMessage = i18n.translate('xpack.ml.fileDatavisualizer.editFlyout.overrides.customTimestampValidationErrorMessage', {
+      defaultMessage: 'Timestamp format {timestampFormat} not supported because it contains a question mark character ({fieldPlaceholder})',
```

Comment:
There is a disconnect between the `fieldPlaceholder` and the use of `question mark character` in the message - `INDETERMINATE_FIELD_PLACEHOLDER` is passed in but then you have a hardcoded `question mark` in the text. If the `INDETERMINATE_FIELD_PLACEHOLDER` is never likely to change, I'd suggest removing `fieldPlaceholder` as a separate variable and just using `?` directly in the `defaultMessage`. If it is likely to change, maybe we need a separate parameter in the translation for the description of the character.

Reply:
Addressed in 51c237651d7a6baebf826a47672823461f8055d9